processing/src/main/java/org/apache/druid/query/aggregation/mean/DoubleMeanVectorAggregator.java

```diff
@@ -44,8 +44,15 @@ public void init(final ByteBuffer buf, final int position)
   @Override
   public void aggregate(final ByteBuffer buf, final int position, final int startRow, final int endRow)
   {
+    final boolean[] nullVector = selector.getNullVector();
+
     final double[] vector = selector.getDoubleVector();
     for (int i = startRow; i < endRow; i++) {
+      if (nullVector != null) {
+        if (nullVector[i]) {
+          continue;
+        }
+      }
```

Comment:
nit: might be worth having separate for loops for nullVector != null so that we don't have to check if the value is null every loop iteration

Reply:
done

Reply:
I am also using the table `test` from `SimpleTestIndex`. Code works locally on with and without vectorize and also on setting both true and false for `druid.generic.useDefaultValueForNull`. Working on adding test cases to the files suggested in the review